src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs

```diff
@@ -5949,7 +5964,28 @@ private int GetIndexOfClickedItem()
                 pt = (POINT)pos
             };
 
-            return (int)User32.SendMessageW(this, (User32.WM)LVM.HITTEST, IntPtr.Zero, ref lvhi);
+            return lvhi;
+        }
+
+        private int UpdateGroupCollapse(Interop.User32.WM clickType)
+        {
+            var lvhi = SetUpHitTestInfo();
+            // see the mouse is on a group
+            int index = (int)User32.SendMessageW(this, (User32.WM)LVM.SUBITEMHITTEST, (IntPtr)(-1), ref lvhi);
```

Comment:
```suggestion
            // see the mouse is on a group
            var lvhi = SetUpHitTestInfo();
            int index = (int)User32.SendMessageW(this, (User32.WM)LVM.SUBITEMHITTEST, (IntPtr)(-1), ref lvhi);

```

Reply:
I think it needs to be this:
```suggestion
            var lvhi = SetUpHitTestInfo();
            // see the mouse is on a group
            int index = unchecked((int)(long)User32.SendMessageW(this, (User32.WM)LVM.SUBITEMHITTEST, (IntPtr)(-1), ref lvhi));
```

@JeremyKuhne we have several versions of such casts all over the codebase. E.g.
```cs
(int)User32.SendMessageW(...)
(int)(long)User32.SendMessageW(...)
unchecked((int)(long)User32.SendMessageW(...)
PARAM.ToInt(User32.SendMessageW(...)
```

What's the right one? And we need standardize the cast too.

Reply:
`checked` and `unchecked` have no impact on the casts to `IntPtr` as they're actually explicit cast operators defined on the `IntPtr` struct, which does an _explicit_ `checked` when converting to `int` on x64. `(int)(long)` doesn't need `unchecked` as we're not opting out of the default `unchecked` compiler behavior.

The technically "right" one is `(int)(long)User32.SendMessageW(...)`.  That said, this particular usage should never fail as it _should_ never return something that can't fit into an `int`.

I believe, however, that we should be doing #2069 and implicitly handling this ourselves. `PARAM.ToInt` was a intermediate step towards that discussion resolving. In this case it should be `LRESULT.ToInt` or, better yet, per #2069:

``` C#
public struct LRESULT
{
    private IntPtr _value;

    // I prefer implicit, but I can live with these all being explicit
    public static implicit operator int(LRESULT value) => (int)value._value.ToInt64();

    // Other operators
}

// Interop
[DllImport(Libraries.User32, ExactSpelling = true)]
public static extern LRESULT SendMessageW(/* ... */);

// Usage
int index = User32.SendMessage(/* ... */);
```

Reply:
Great, thanks!

In this case I propose we go with my proposal for consistency and for the future ease of refactoring (so we can search ad replace the lot in one go).

`unchecked((int)(long)...` - 130 hits
`(int)(long)` - 16 hits
`(int)SendMessage` - 2 hits

Reply:
/cc @hughbe 😉